### PR TITLE
Configure muzzle plugins

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
(part 1)

### CODE REVIEW

**Comment by mateuszrzeszutek** [comment]
@iNikem please take a look at this PR, whether it fits your idea of using muzzle in our project

File: instrumentation/khttp/build.gradle.kts [hunk 1]
```diff
@@ -9,6 +10,15 @@ repositories {
   }
 }
 
+muzzle {
+  pass {
+    group.set("com.github.jkcclemens")
+    module.set("khttp")
+    versions.set("(,)")
+    assertInverse.set(true)
```
Comment (iNikem) [inline instrumentation/khttp/build.gradle.kts:18]: This line probably does not make sense here

File: instrumentation/servlet/build.gradle.kts [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+    group.set("javax.servlet")
+    module.set("servlet-api")
+    versions.set("[2.3,)")
+    assertInverse.set(true)
+  }
 }
 
 dependencies {
-  compileOnly("javax.servlet:servlet-api:2.2")
+  compileOnly("javax.servlet:servlet-api:2.3")
```
Comment (iNikem) [inline instrumentation/servlet/build.gradle.kts:16]: Why this?

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java [hunk 3]
```diff
@@ -33,8 +33,8 @@
 
   private static Stream<Arguments> supportedConfigurations() {
     return configurations("payara")
-        .otelLinux("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11")
-        .otelWindows("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11")
+        .otelLinux("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_HOTSPOT, "8")
+//        .otelWindows("5.2020.6", PAYARA_SERVER_ATTRIBUTES, VMS_ALL, "8", "11")
```
Comment (iNikem) [inline smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:37]: ?

**Review by iNikem — COMMENTED** [review]
Until we start publishing `io.opentelemetry.instrumentation.javaagent-instrumentation`, this is good enough, yeah :)

File: instrumentation/servlet/build.gradle.kts (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/servlet/build.gradle.kts:16]: We check for `javax.servlet.Filter` class in the `classLoaderMatcher()`, that class was added in 2.3.

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:37]: Remains of local testing, I forgot to remove that - thanks!

**Review by iNikem — APPROVED** [review]

**Pull request merged**